runtime/src/main/kotlin/kotlin/native/internal/Annotations.kt

```diff
@@ -104,4 +104,12 @@ internal annotation class PointsTo(vararg val onWhom: Int)
 
 @Target(AnnotationTarget.FUNCTION)
 @Retention(AnnotationRetention.BINARY)
-internal annotation class TypedIntrinsic(val kind: String)
\ No newline at end of file
+internal annotation class TypedIntrinsic(val kind: String)
+
+/**
+ * Indicates that `@SymbolName external` function is implemented in library-stored bitcode
+ * and doesn't have native dependencies.
+ */
+@Target(AnnotationTarget.FUNCTION)
+@Retention(AnnotationRetention.BINARY)
+annotation class Independent
```

Comment:
Not sure if the name is too enlightening, what about smth like `LocallyDefined`? Also what confuses me, is that functions in bitcode usually call code in the native library, why can we ignore that? Or it's only applied to declarations without real external deps?

Reply:
I agree that the name is confusing, but the semantics is too subtle. `LocallyDefined` is nice, but does it explain why Objective-C class member stubs are locally defined, while category member stubs aren't? Bodies of stubs of both kinds are indistinguishable.

> Or it's only applied to declarations without real external deps?

Exactly.